src/tests/migration.test.ts

```diff
@@ -199,3 +190,173 @@ describe('_20230124_164900', () => {
+      explorerUrl: 'https://explorer-v18.testnet.alephium.org'
+    })
+
+    migrate._v150_networkSettingsMigration()
+
+    const migratedSettings = SettingsStorage.load('network') as NetworkSettings
+
+    expect(migratedSettings.nodeHost).toEqual(networkPresets.testnet.nodeHost)
+    expect(migratedSettings.explorerApiHost).toEqual(networkPresets.testnet.explorerApiHost)
+    expect(migratedSettings.explorerUrl).toEqual(networkPresets.testnet.explorerUrl)
+  })
+
+  it('should not migrate pre-v1.5.0 custom network settings', () => {
+    const customSettings = {
+      networkId: 3,
+      nodeHost: 'https://mainnet-wallet.custom.com',
+      explorerApiHost: 'https://mainnet-backend.custom.com',
+      explorerUrl: 'https://explorer.custom.com'
+    }
+
+    SettingsStorage.store('network', customSettings)
+
+    migrate._v150_networkSettingsMigration()
+
+    const migratedSettings = SettingsStorage.load('network') as NetworkSettings
+
+    expect(migratedSettings.nodeHost).toEqual(customSettings.nodeHost)
+    expect(migratedSettings.explorerApiHost).toEqual(customSettings.explorerApiHost)
+    expect(migratedSettings.explorerUrl).toEqual(customSettings.explorerUrl)
+  })
+})
+
+describe('_20230209_124300', () => {
+  const addressesMetadataLocalStorageKeyPrefix = 'addresses-metadata'
+  const addressesMetadataEncryptionVersion = '2022-05-27T12:00:00Z'
+  const wallet = {
+    name: 'John Doe',
+    unencryptedWallet: walletGenerate()
+  }
+
+  it('should replace the isMain address metadata settings with isDefault and ensure there is a color', () => {
+    const dataKey = { mnemonic: wallet.unencryptedWallet.mnemonic, walletName: wallet.name }
+    const deprecatedAddressMetadata: DeprecatedAddressMetadata[] = [
+      {
+        index: 0,
+        isMain: false
+      },
+      {
+        index: 1,
+        isMain: true,
+        color: 'red',
+        label: 'My main one'
+      }
+    ]
+
+    localStorage.setItem(
+      `${addressesMetadataLocalStorageKeyPrefix}-${stringToDoubleSHA256HexString(wallet.name)}`,
+      JSON.stringify({
+        version: addressesMetadataEncryptionVersion,
+        encryptedSettings: encrypt(wallet.unencryptedWallet.mnemonic, JSON.stringify(deprecatedAddressMetadata))
+      })
+    )
+
+    migrate._20230209_124300(dataKey)
+
+    const metadata = AddressMetadataStorage.load(dataKey) as AddressMetadata[]
+
+    expect(Object.prototype.hasOwnProperty.call(metadata[0], 'isMain')).toBeFalsy()
+    expect(metadata[0].isDefault).toEqual(false)
+    expect(metadata[0].index).toEqual(0)
+    expect(metadata[0].color).toBeDefined()
+    expect(metadata[0].color).toBeTruthy()
+    expect(Object.prototype.hasOwnProperty.call(metadata[1], 'isMain')).toBeFalsy()
+    expect(metadata[1].isDefault).toEqual(true)
+    expect(metadata[1].index).toEqual(1)
+    expect(metadata[1].color).toEqual('red')
+    expect(metadata[1].label).toEqual('My main one')
+  })
+})
```

Comment:
I've done a lot of work to migrate our storage:
1. Created release-specific network settings migrations
2. Created migration to change `isMain` to `isDefault` in address metadata
3. Created migration to ensure that every address has a color